src/src/beneficiaries/beneficiaries.service.ts

```diff
@@ -172,6 +179,166 @@ export class BeneficiariesService {
+			if (!user?.data?.program_users?.[0]?.organisation_id) {
+				return resp.status(404).send({
+					success: false,
+					message: 'Invalid Ip',
+					data: {},
+				});
+			}
+			const sortType = body?.sortType ? body?.sortType : 'desc';
+			let status = body?.status;
+			let filterQueryArray = [];
+			filterQueryArray.push(
+				`{ program_beneficiaries: { facilitator_user: { program_faciltators: { parent_ip: { _eq: "${user?.data?.program_users[0]?.organisation_id}" } } } } }`,
+			);
+			if (body?.district && body?.district.length > 0) {
+				filterQueryArray.push(
+					`{district:{_in: ${JSON.stringify(body?.district)}}}`,
+				);
+			}
+
+			if (body?.block && body?.block.length > 0) {
+				filterQueryArray.push(
+					`{block:{_in: ${JSON.stringify(body?.block)}}}`,
+				);
+			}
+
+			if (body.facilitator && body.facilitator.length > 0) {
+				filterQueryArray.push(
+					`{program_beneficiaries: {facilitator_id:{_in: ${JSON.stringify(
+						body.facilitator,
+					)}}}}`,
+				);
+			}
+
+			if (status && status !== '') {
+				if (status === 'identified') {
+					filterQueryArray.push(`{
+					_or: [
+						{ program_beneficiaries: { status: { _eq: "identified" } } },
+						{ program_beneficiaries: { status: { _is_null: true } } },
+						{ program_beneficiaries: { status: { _eq: "" } } },
+					]
+				}`);
+				} else {
+					filterQueryArray.push(
+						`{program_beneficiaries:{status:{_eq:${status}}}}`,
+					);
+				}
+			}
+
+			let filterQuery = '{ _and: [' + filterQueryArray.join(',') + '] }';
+			var data = {
+				query: `query MyQuery {
+				users(where: ${filterQuery},
+                      order_by: {
+                        created_at: ${sortType}
+                      }
+				) {
+					id
+					first_name
+					last_name
+					district
+					block
+					mobile
+				    program_beneficiaries {
+					id
+					enrolled_for_board
+					subjects
+					facilitator_id
+					status
+					
+				  }
+				}
+			  }`,
+			};
+			const response = await this.hasuraServiceFromServices.getData(data);
+			let result = response?.data?.users;
+			let mappedResponse = result;
+			console.log(
+				'mappedResponse',
+				mappedResponse[1],
+				mappedResponse.length,
+			);
+
+			const sql = `SELECT
+						name,
+						array_agg(id)
+						FROM
+						subjects
+						GROUP BY
+						name
+						;`;
+			const subjectGroup = (
+				await this.hasuraServiceFromServices.executeRawSql(sql)
+			).result;
+			let allSubjects =
+				this.hasuraServiceFromServices.getFormattedData(subjectGroup);
+			const subjects = {};
+			const subjectHeader = [];
+			for (let data of allSubjects) {
+				subjectHeader.push({ id: data.name, title: data.name });
+			}
+
+			const csvStringifier = createObjectCsvStringifier({
+				header: [
+					{ id: 'name', title: 'Name' },
+					{ id: 'enrolled_for_board', title: 'Enrolled For Board' },
+					...subjectHeader,
+				],
+			});
+			const records = [];
+			for (let data of mappedResponse) {
```

Comment:
add empty line before and after each `if`, `for`

Reply:
aded